### ISSUE TITLE
Sqlfluffignore file isn't read if it is located at the filesystem root of a Linux system

### ISSUE THREAD

**Issue by baa-ableton** [sqlfluff/sqlfluff#4181]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

We lint our codebase using sqlfluff. We deploy our entire codebase in the root of the filesystem of a docker container during CI\CD. We have one, master .sqlfluffignore that resides at the root of the project, consequently, this .sqlfluffignore is also located at the root of the filesystem.

The function paths_from_path in linter.py adds an os.sep character at the end of the base where .sqlfluffignore is found which is the way to go in almost all cases except for when the .sqlfluffignore file is found at the root. Then, all gitignore-like path specifications to ignore are built from '//.sqlfulffignore'. 

### Expected Behaviour

````bash
root@e8d186c674fb:/# sqlfluff lint --format human sql/some/path/00_copy_logs_from_s3.sql 
WARNING    Exact file path sql/some/path/00_copy_logs_from_s3.sql was given but it was ignored by a .sqlfluffignore pattern in /, re-run with `--disregard-sqlfluffignores` to skip .sqlfluffignore 
All Finished 📜 🎉!
root@e8d186c674fb:/# 
````

### Observed Behaviour

````bash
root@e8d186c674fb:/# sqlfluff lint --format human sql/some/path/00_copy_logs_from_s3.sql 
matching:   0%|                                                                                                                                                                                                             | 0/3467 [00:00<?, ?it/s]
````

### How to reproduce

Dockerfile: 
````Dockerfile
FROM python:3

COPY . .

RUN mkdir -p some/path/ &&\
         mv test.sql some/path/

RUN pip install sqlfluff

CMD ["sqlfluff lint --format human some/path/test.sql"]
````

### Dialect

Redshift

### Version

1.4.1

### Configuration


### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
